qa-core/src/tests/internal-api/screens/screens.spec.ts

```diff
@@ -0,0 +1,71 @@
+import TestConfiguration from "../../../config/internal-api/TestConfiguration"
+import { App } from "@budibase/types"
+import InternalAPIClient from "../../../config/internal-api/TestConfiguration/InternalAPIClient"
+import generateApp from "../../../config/internal-api/fixtures/applications"
+import { Screen } from  "@budibase/types"
+import generateScreen from "../../../config/internal-api/fixtures/screens"
+
+
+describe("Internal API - /screens endpoints", () => {
+  const api = new InternalAPIClient()
+  const config = new TestConfiguration<Screen>(api)
+  const appConfig = new TestConfiguration<App>(api)
+
+  beforeAll(async () => {
+    await config.beforeAll()
+  })
+
+  afterAll(async () => {
+    await config.afterAll()
+  })
+
+  it("POST - Create a screen with each role type", async () => {
+    // Create app
+    const [appResponse, app] = await appConfig.applications.create(generateApp())
+    expect(appResponse).toHaveStatusCode(200)
+    expect(app._id).toBeDefined()
+    
+    // Create Screen
+    const roleArray = ["BASIC", "POWER", "ADMIN", "PUBLIC"]
```

Comment:
An alternative for this could be to select a random role when a screen gets created (via `/fixtures/screens.ts`)

In theory, with this test running frequently, we would always get the same coverage of all role types without having to specify anytime we call `generateScreen`

Just curious of any thoughts on this?